### PR TITLE
type: [feat], message: Add commit lint option.

### PR DESCRIPTION
**Pull request by esequielvirtuoso**
## What?

type: [feat], message: Add commit lint option.

### Type of change?

- [x] [feat]: A new feature

Note: Remove the non-checked options

## Why?

Here you express the reason it is necessary (keep the _so what?_ factor in mind)

## How?

Here you give a little more detail on the implementation or justify any unorthodox choice or practice.

## How has this been tested?

- [x] Unit tests

# Before submitting a PR please make sure to check if:

- [x] The code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] The changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and/or existing unit tests pass locally with my changes
- [x] New and/or existing integration tests pass locally with my changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,6 +76,32 @@ setup(
     install_requires=requirements,
     packages=find_packages(),
 )
+```
+
+ ### How to add commit lint stage to Gitlab?
+
+ You must add a new stage to `gitlab-ci.yml` file adding two new arguments to semantic-release script.
+ - `commit-lint=true` to run commit-lint logic;
```
Comment (julianogalgaro) [inline README.md:84]:
```suggestion
 - `-commit-lint=true` to run commit-lint logic;
```
@esequielvirtuoso

File: cmd/semantic-release/semantic-release.go [hunk 2]
```diff
@@ -67,15 +70,27 @@ func main() {
 	case "up":
 		logger.Info(colorYellow + "\nSemantic Version just started the process...\n\n" + colorReset)
 
-		semantic := newSemantic(logger, upgradeVersionCmd, gitHost, groupName, projectName, username, password, upgradePyFile)
-
-		if err := semantic.GenerateNewRelease(); err != nil {
-			logger.Error(err.Error())
-			os.Exit(1)
+		fmt.Println(*gitHost)
```
Comment (julianogalgaro) [inline cmd/semantic-release/semantic-release.go:73]: should we change to logger? Maybe we can move it to the previous log Info.

**Review by julianogalgaro — APPROVED** [review]
nice!

File: cmd/semantic-release/semantic-release.go (hunk 2 above)
Comment (esequielvirtuoso) [reply cmd/semantic-release/semantic-release.go]: @julianogalgaro 

I've removed it.

It was a useless print.

File: README.md (hunk 1 above)
Comment (esequielvirtuoso) [reply README.md]: @julianogalgaro 
Added!

File: hack/set-pre-commit.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+mkdir -p ~/.git-templates/hooks
+cp -f ./hack//githooks/commit-msg ~/.git-templates/hooks
```
Comment (ebbauer) [inline hack/set-pre-commit.sh:3]: @esequielvirtuoso why the double slash?

File: cmd/semantic-release/semantic-release.go [hunk 4]
```diff
@@ -67,15 +70,26 @@ func main() {
...
+
+		if *commitLint {
+			if *branchName == "" {
+				logger.Error(colorRed + "\nThe argument -branch-name must be set when --commit-lint is true.\n\n" + colorReset)
+			}
+
+			logger.Info(colorYellow + "\nSemantic Version commit lint started...\n\n" + colorReset)
+			err := semantic.CommitLint()
+			if err != nil {
+				os.Exit(1)
```
Comment (ebbauer) [inline cmd/semantic-release/semantic-release.go:83]: @esequielvirtuoso should we log the error message?

**Review by ebbauer — COMMENTED** [review]
some files have different naming pattern than others

File: cmd/semantic-release/semantic-release.go (hunk 4 above)
Comment (esequielvirtuoso) [reply cmd/semantic-release/semantic-release.go:86]: @ebbauer 
We're logging it at the `semantic` package.

File: hack/set-pre-commit.sh (hunk 3 above)
Comment (esequielvirtuoso) [reply hack/set-pre-commit.sh]: @ebbauer 

Don't worry, this file has been removed.

**Review by ebbauer — APPROVED** [review]

**Review by julianogalgaro — APPROVED** [review]
nice, but maybe we need to update the mr description with the new pattern.

**Pull request merged**